src/backend/distributed/operations/shard_rebalancer.c

```diff
@@ -2609,7 +2611,17 @@ ActivePlacementsHash(List *shardPlacementList)
 	foreach(shardPlacementCell, shardPlacementList)
 	{
 		ShardPlacement *shardPlacement = (ShardPlacement *) lfirst(shardPlacementCell);
-		if (shardPlacement->shardState == SHARD_STATE_ACTIVE)
+		WorkerNode *workerNode = FindWorkerNode(shardPlacement->nodeName,
+												shardPlacement->nodePort);
+
+		/*
+		 * If a worker can be found for the shard, that worker should be active.
+		 * For the sake of testing rebalance and replicate, having a null
+		 * worker is also allowed since the tests are done with dummy shard placements
+		 * and workers. The worker will not be NULL with actual shard placements.
+		 */
+		if (shardPlacement->shardState == SHARD_STATE_ACTIVE &&
```

Comment:
Also, do we need any improvements in `ReplicationPlacementUpdates`, where we have:

```C
if (placement->shardState != SHARD_STATE_ACTIVE)
		{
			continue;
		}

```

Should that check for placements on inactive nodes:
```C
if (placement->shardState != SHARD_STATE_ACTIVE || PlacemenOnInactiveNode(placement))
		{
			continue;
		}
```

Reply:
This doesn't matter. Since we create a table to count how many shard do all the workers have, and all the workers passed to this function are active. Then even though we don't skip the placements in inactive workers, they do not contribute to the count of the workers.